core/src/mos6502.cpp

```diff
@@ -110,115 +83,132 @@ void Mos6502::execute() {
+                pipeline_.push([=]() { ++registers_->pc; });
+                pipeline_.push(store_byte_abs_addr(registers_->y));
+                return;
+            }
+            break;
+        case Instruction::STA:
+            if (opcode.addressMode == AddressMode::Absolute) {
+                pipeline_.push([=]() { ++registers_->pc; });
+                pipeline_.push([=]() { ++registers_->pc; });
+                pipeline_.push(store_byte_abs_addr(registers_->a));
+                return;
+            }
+            break;
+        case Instruction::STX:
+            if (opcode.addressMode == AddressMode::Absolute) {
+                pipeline_.push([=]() { ++registers_->pc; });
+                pipeline_.push([=]() { ++registers_->pc; });
+                pipeline_.push(store_byte_abs_addr(registers_->x));
+                return;
+            }
+            break;
+        case Instruction::BCC:
             pipeline_.push(
                     branch_on([=]() { return !(registers_->p & C_FLAG); }));
             return;
-        case LDY_I:
-            pipeline_.push([=]() {
-                registers_->y = mmu_->read_byte(registers_->pc++);
-                set_zero(registers_->y);
-                set_negative(registers_->y);
-            });
-            return;
-        case BCS:
+        case Instruction::LDY:
+            if (opcode.addressMode == AddressMode::Immediate) {
+                pipeline_.push([=]() {
+                    registers_->y = mmu_->read_byte(registers_->pc++);
+                    set_zero(registers_->y);
+                    set_negative(registers_->y);
+                });
+                return;
+            }
+            break;
+        case Instruction::BCS:
             pipeline_.push(branch_on([=]() { return registers_->p & C_FLAG; }));
             return;
-        case CLV:
+        case Instruction::CLV:
             pipeline_.push([=]() { clear_flag(V_FLAG); });
             return;
-        case BNE:
+        case Instruction::BNE:
             pipeline_.push(
                     branch_on([=]() { return !(registers_->p & Z_FLAG); }));
             return;
-        case CLD:
+        case Instruction::CLD:
             pipeline_.push([=]() { clear_flag(D_FLAG); });
             return;
-        case NOP:
+        case Instruction::NOP:
             pipeline_.push([]() { /* Do nothing. */ });
             return;
-        case INX:
+        case Instruction::INX:
             pipeline_.push([=]() {
                 ++registers_->x;
                 set_zero(registers_->x);
                 set_negative(registers_->x);
             });
             return;
-        case BEQ:
+        case Instruction::BEQ:
             pipeline_.push(branch_on([=]() { return registers_->p & Z_FLAG; }));
             return;
-        case SED:
+        case Instruction::SED:
             pipeline_.push([=]() { set_flag(D_FLAG); });
             return;
+        case Instruction::Invalid:
+        default:
```

Comment:
The reason I omitted default was so that any opcodes added to the enum would warn and error here instead of only exploding at run-time. It looks like that would still work?

Reply:
Ah, good catch!